scripts/models/color-token/color-token.js

```diff
@@ -0,0 +1,197 @@
+const Token = require("../token");
+
+/**
+ * Token used to represent a color token collection.
+ */
+class ColorToken extends Token {
+  /**
+   * Construct a new ColorToken.
+   *
+   * @param {Token.Config} dto - ColorToken configuration Object.
+   */
+  constructor(config) {
+    super({ category: Token.CONSTANTS.CATEGORIES.COLOR, ...config });
+  }
+
+  /**
+   * Update the format of this ColorToken.
+   *
+   * @param {DTO} dto
+   * @param {Token.Format} dto.format - The format to update this ColorToken to.
+   * @returns {this} - This ColorToken.
+   */
+  updateFormat({ format }) {
+    const final = ColorToken.translateColors({ from: this.format, to: format, colors: this.data });
+    this.format = format;
+    this.data = final;
+
+    return this;
+  }
+
+  /**
+   * Normalizes this ColorToken into the project's Standard format.
+   *
+   * @returns {this} - This color token.
+   */
+  normalize() {
+    return this.updateFormat({ format: Token.CONSTANTS.TOKEN_FORMATS.STANDARD });
+  }
+
+  /**
+   * Constants associated with the ColorToken Object.
+   */
+  static get CONSTANTS() {
+    return { ...Token.CONSTANTS };
+  }
+
+  /**
+   * Merge ColorToken Grades into an existing ColorToken's Grades.
+   *
+   * @param {DTO} dto
+   * @param {ColorToken.Grades} dto.source - ColorToken Grades to merge into the destination ColorToken Grades.
+   * @param {ColorToken.Grades} dto.destination - ColorToken Grades to be receiving a merge.
+   * @returns {ColorToken.Grades} - Merged ColorToken Grades.
+   */
+  static mergeGrades({ source, destination }) {
+    return Object.entries(source).reduce((final, [grade, value]) => ({ ...final, [grade]: value }), { ...destination });
+  }
+
+  /**
+   * Merge ColorToken Colors into an existing ColorToken's Colors.
+   *
+   * @param {DTO} dto
+   * @param {ColorToken.Colors} dto.source - ColorToken Colors to merge into the destination ColorToken Colors.
+   * @param {ColorToken.Colors} dto.destination - ColorToken Colors to be receiving a merge.
+   * @returns {ColorToken.Colors} - Merged ColorToken Colors.
+   */
+  static mergeColors({ source, destination }) {
+    return Object.entries(source).reduce(
+      (final, [color, value]) => ({
+        ...final,
+        [color]: ColorToken.mergeGrades({ source: value, destination: destination[color] }),
+      }),
+      { ...destination }
+    );
+  }
+
+  /**
+   * Merge a ColorToken into an existing ColorToken.
+   *
+   * @param {DTO} dto
+   * @param {ColorToken} dto.source - ColorToken to merge into the destination ColorToken.
+   * @param {ColorToken} dto.destination - ColorToken to be receiving a merge.
+   * @returns {ColorToken} - Merged ColorToken.
```

Comment:
probably worth adding here that the returned ColorToken is normalised

Reply:
Will do! Updating the JSDoc now.